opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/mod.rs

```diff
@@ -1 +1,222 @@
-mod lz4_chunked_compression;
+pub(crate) mod bond_encoder;
+pub(crate) mod central_blob;
+pub(crate) mod lz4_chunked_compression;
+pub(crate) mod otlp_encoder;
+
+#[cfg(test)]
+mod tests {
+    use crate::payload_encoder::bond_encoder::{BondDataType, BondEncodedSchema, BondWriter};
+    use crate::payload_encoder::central_blob::{
+        CentralBlob, CentralEventEntry, CentralSchemaEntry,
+    };
+
+    use crate::payload_encoder::bond_encoder::FieldDef;
+    use std::borrow::Cow;
+
+    fn create_payload(fields: &[(&str, u8, u16)], row_data: Vec<u8>) -> Vec<u8> {
+        // Convert the input tuples to FieldDef
+        let field_defs: Vec<FieldDef> = fields
+            .iter()
+            .map(|(name, type_id, field_id)| FieldDef {
+                name: Cow::Owned(name.to_string()),
+                type_id: *type_id,
+                field_id: *field_id,
+            })
+            .collect();
+        let schema_obj =
+            BondEncodedSchema::from_fields("MdsContainer", "testNamespace", field_defs);
+        let schema_bytes = schema_obj.as_bytes();
+        let schema_md5 = md5::compute(schema_bytes).0;
+        let schema_id = 1u64;
+
+        let schema_entry = CentralSchemaEntry {
+            id: schema_id,
+            md5: schema_md5,
+            schema: schema_obj,
+        };
+
+        let event = CentralEventEntry {
+            schema_id,
+            level: 5,
+            event_name: "basename".to_string(),
+            row: row_data,
+        };
+
+        let metadata =
+            "namespace=testNamespace/eventVersion=Ver1v0/tenant=T/role=R/roleinstance=RI";
+
+        let blob = CentralBlob {
+            version: 1,
+            format: 2,
+            metadata: metadata.to_string(),
+            schemas: vec![schema_entry],
+            events: vec![event],
+        };
+
+        blob.to_bytes()
+    }
+
+    #[test]
+    #[allow(clippy::approx_constant)]
+    fn test_single_column_double() {
+        // The encoded value using official C++ Bond encoder
+        let expected: &[u8] = &[
```

Comment:
We could explore the possibility of using deserialization through the C++ package just for the tests. That would allow us to do typed value checks instead of comparing bytes.